src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs

```diff
@@ -112,44 +112,29 @@ public unsafe static uint[] Square(uint[] value)
                 Square(valueHigh, valueHighLength,
                        bitsHigh, bitsHighLength);
 
-                // ... compute z_a = a_1 + a_0 (call it fold...)
                 int foldLength = valueHighLength + 1;
-                uint* fold = stackalloc uint[foldLength];
-
-                Add(valueHigh, valueHighLength,
-                    valueLow, valueLowLength,
-                    fold, foldLength);
-
-                // ... compute z_1 = z_a * z_a - z_0 - z_2
                 int coreLength = foldLength + foldLength;
-                uint* core = stackalloc uint[coreLength];
-
-                Square(fold, foldLength,
-                       core, coreLength);
-                SubtractCore(bitsHigh, bitsHighLength,
-                             bitsLow, bitsLowLength,
-                             core, coreLength);
-
-                // ... and finally merge the result! :-)
-                AddSelf(bits + n, bitsLength - n, core, coreLength);
+                fixed (uint* fold = new uint[foldLength],
```

Comment:
I didn't entirely follow the conversation between you and @KrzysztofCwalina... why did we change the existing stackallocs to instead be array allocations with fixed?